### PR TITLE
Add groupName and groupPassword arguments to CodingDojo.new

### PR DESCRIPTION
**Pull request by totalgee**
- this allows you to set up different rooms/groups instead of being
  stuck with the default of 'oscrouter'.
- (resolves issue #1)

### CODE REVIEW

File: Classes/CodingDojo.sc [hunk 1]
```diff
@@ -1,19 +1,21 @@
 CodingDojo {
-	var <username, <password, <serveraddress, <serverport;
+	var <username, <password, <serveraddress, <serverport, <groupName, <groupPassword;
```
Comment (adcxyz) [inline Classes/CodingDojo.sc:2]: groupName and password can just live in oscrouter, copies here can go out of sync.

File: Classes/CodingDojo.sc [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 	var <>turnTime = 300, <remainTime, <timer;
 	var <pilot, <copilot, <nextCopilot, <myStatus, <order;
 	var <win, <uv;
 
-	*new { arg username, password, serveraddress = "bgo.la", serverport = 55555;
-		^super.newCopyArgs(username.asSymbol, password.asSymbol, serveraddress.asString, serverport).init;
+	*new { arg username, password, serveraddress = "bgo.la", serverport = 55555,
+		groupName, groupPassword;
+		^super.newCopyArgs(username.asSymbol, password.asSymbol, serveraddress.asString, serverport, groupName, groupPassword).init;
 	}
```
Comment (adcxyz) [inline Classes/CodingDojo.sc:11]: if not making making separate vars in CodingDojo, pass group name and password into init

File: Classes/CodingDojo.sc [hunk 3]
```diff
@@ -39,6 +41,8 @@ CodingDojo {
 		syncText.showDoc;
 		this.addOSCFuncs;
 		this.enableCodeSending;
+		groupName = oscrouter.groupName;
+		groupPassword = oscrouter.groupPassword
 	}
```
Comment (adcxyz) [inline Classes/CodingDojo.sc:46]: these could just be accessors into oscrouter:
```groupName { ^oscrouter.groupName }```

File: Classes/OSCRouterClient.sc [hunk 4]
```diff
@@ -27,9 +27,12 @@ OSCRouterClient {
...
 	// use that to avoid doubled login failure and confusion.
 	*new { arg serverAddr, userName, userPassword, onJoined,
-		groupName = 'oscrouter', groupPassword = 'oscrouter', serverport = 55555;
+		groupName, groupPassword, serverport = 55555;
 
-		var found = this.findBy(userName, groupName, serverAddr, serverport);
+		var found;
+		groupName = groupName ? 'oscrouter';
+		groupPassword = groupPassword ? 'oscrouter';
+		found = this.findBy(userName, groupName, serverAddr, serverport);
```
Comment (adcxyz) [inline Classes/OSCRouterClient.sc:36]: I like the solution as is better: this way, the default values are shown in the code completion when writing 
```OSCRouterClient(...)```

**Comment by adcxyz** [comment]
Hi @totalgee,
73703da fixes this according to my review notes, so I close it.
Again, thanks for proposing this!

**Comment by totalgee** [comment]
Thanks, looks good and I just tested it. One question remains -- do we also need to be able to change the "filename" of the `SyncText` object used? Currently it's always set to `CodingDojoSession`...will that "file" conflict with people in other groups doing CodingDojo on the same server, or does each group get its own set of SyncText "filenames"?

Thanks.

**Comment by adcxyz** [comment]
As long as the group names are different, the groups on the server know nothing about each other; so people in group \Berlin only get connected others in \Berlin, never from \Barcelona, 
so there should be no "crosstalk".
I guess being in two CodingDojoSessions in different groups at the same time defeats the purpose of the CodingDojo practice :-) 
Do you have a use case where it makes sense?
CodingDojo Olympics competitions maybe, sort of like simultaneous chess games?

**Pull request closed without merging**